src/MSBuildTaskHost/Immutable/ImmutableDictionary.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace System.Collections.Immutable
```

Comment:
What's wrong with [System.Collections.Immutable.ImmutableDictionary](https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutabledictionary-2?view=netcore-3.1)?

Reply:
Nothing that a time machine wouldn't fix! This PR uses it where it's available, but it is not available in .NET 3.5. Which we use in MSBuildTaskHost.exe, to execute 3.5-only tasks.

So this is an API-compatible (at least what we use) reimplementation. It's inefficient, but I don't care: update your project to .NET 4. It came out in 2012!

We've used this strategy for types in `System.Collections.Concurrent`, too, though I think those might be full source copies.

Reply:
That makes sense, but it deserves 😞